domainlab/arg_parser.py

```diff
@@ -148,8 +153,21 @@ def parse_cmd_args():
     """
     parser = mk_parser_main()
     args = parser.parse_args()
+    if args.config_file:
+
+        data = yaml.safe_load(args.config_file)
+        delattr(args, 'config_file')
+        arg_dict = args.__dict__
+
+        for key, value in data.items():
+            if isinstance(value, list):
+                arg_dict[key].extend(value)
```

Comment:
do we need a warning here that if there exist some items in the list yaml file inputs will further expand the list?  can it happen that the user provided something which already exist in yaml file?

Reply:
I added a check to (1) throw an exception if yaml file contains fields not in argparser.

Reply:
I added a check to (1) throw an exception if yaml file contains fields not in argparser.

> do we need a warning here that if there exist some items in the list yaml file inputs will further expand the list? can it happen that the user provided something which already exist in yaml file?

I'm not sure about this. I think users would either want to use a yaml file or just command-line arguments. I don't see a reason as to why one would choose both. In the current setup, the parameters in the yaml file would overwrite any command-line arguments if they are both the same.